### PR TITLE
implemented controllers and validators password and email

### PR DESCRIPTION
**Pull request by Daniel1004K**

### CODE REVIEW

File: src/main/java/mate/academy/spring/controller/UserController.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            DtoResponseMapper<UserResponseDto, User> userDtoResponseMapper) {
+        this.userService = userService;
+        this.userDtoResponseMapper = userDtoResponseMapper;
+    }
+
+    @GetMapping("/by-email")
+    public UserResponseDto getByEmail(@RequestParam String email) {
+        return userDtoResponseMapper.toDto(userService.findByEmail(email).get());
+    }
+
```
Comment (nikdagaev) [inline src/main/java/mate/academy/spring/controller/UserController.java:28]: redundant empty line

**Review by nikdagaev — CHANGES_REQUESTED** [review]

**Review by nikdagaev — APPROVED** [review]

File: src/main/java/mate/academy/spring/controller/GlobalExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import org.springframework.validation.FieldError;
+import org.springframework.validation.ObjectError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class GlobalExceptionHandler extends ResponseEntityExceptionHandler {
```
Comment (Sofasmile) [inline src/main/java/mate/academy/spring/controller/GlobalExceptionHandler.java:23]: this code doesn't relate to current HW, we will do it in the next one, let's not push currently unneeded code in future

File: src/main/java/mate/academy/spring/controller/OrderController.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    @PostMapping("/complete")
+    public OrderResponseDto completeOrder(@RequestParam Long userId) {
+        ShoppingCart shoppingCart = shoppingCartService.getByUser(
+                userService.finById(userId));
+        return orderResponseMapper.toDto(orderService.completeOrder(shoppingCart));
+    }
+
+    @GetMapping("/orders")
```
Comment (Sofasmile) [inline src/main/java/mate/academy/spring/controller/OrderController.java:44]:
```suggestion
    @GetMapping
```

File: src/main/java/mate/academy/spring/model/dto/request/CinemaHallRequestDto.java [hunk 4]
```diff
@@ -1,7 +1,11 @@
 package mate.academy.spring.model.dto.request;
 
+import javax.validation.constraints.NotNull;
+
 public class CinemaHallRequestDto {
+    @NotNull
```
Comment (Sofasmile) [inline src/main/java/mate/academy/spring/model/dto/request/CinemaHallRequestDto.java:6]: can int be null?

File: src/main/java/mate/academy/spring/model/dto/request/UserRegistrationRequestDto.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
+package mate.academy.spring.model.dto.request;
+
+import mate.academy.spring.service.validation.Email;
+import mate.academy.spring.service.validation.Password;
+
+public class UserRegistrationRequestDto {
+    @Email
+    private String email;
+    @Password
```
Comment (Sofasmile) [inline src/main/java/mate/academy/spring/model/dto/request/UserRegistrationRequestDto.java:9]: we will do it in the next hw

File: src/main/java/mate/academy/spring/service/UserService.java [hunk 6]
```diff
@@ -7,4 +7,6 @@
     User add(User user);
 
     Optional<User> findByEmail(String email);
+
+    User finById(Long id);
```
Comment (Sofasmile) [inline src/main/java/mate/academy/spring/service/UserService.java:11]:
```suggestion
    User findById(Long id);
```

File: src/main/java/mate/academy/spring/service/validation/EmailValidator.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
+package mate.academy.spring.service.validation;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class EmailValidator implements ConstraintValidator<Email, String> {
```
Comment (Sofasmile) [inline src/main/java/mate/academy/spring/service/validation/EmailValidator.java:8]: now I won't review this part of hw

**Review by Sofasmile — CHANGES_REQUESTED** [review]

**Pull request closed without merging**